### ISSUE TITLE
Highlighted ident guide error?

### ISSUE THREAD

**Issue by augustocdias** [Microsoft/vscode#49342]
Shouldn't the highlighted guide be the one with the blue arrow instead the one with the red when the cursor is at the green line?
![screen shot 2018-05-07 at 09 39 40](https://user-images.githubusercontent.com/1080789/39690770-88f27548-51db-11e8-98b2-11436a38c639.png)

**Comment by Jupakabra** [comment]
It's debatable, if you put cursor at `"**/*.txt"` you expect it to mark it's parent indent which is `"todohighlight.include"`, and in this case you are marking `"todohighlight.include"` who's parent is whatever is at the top level, so it work correctly.

One could argue that if selected child node has nodes it's children indent should be selected instead.

**Comment by augustocdias** [comment]
I agree that it is debatable... In my point of view if the cursor is after the block start it should move the highlighted guide to the next one. On the other hand, if the cursor is before, I agree with you @Jupakabra

**Comment by alexandrudima** [comment]
Another example from @Tresky in #49384

### Sample Code
```
function foo (cb) { // this scope gets highlighted
|   // placing cursor on if-statement on the below line will highlight the tab for the function-scope; expected the if-scope.
|   if (this.bar) { // < --- this tab-line *should* be highlighted
|       cb()
|   }
}
```

### Screenshot
<img width="461" alt="screen shot 2018-05-07 at 1 26 05 pm" src="https://user-images.githubusercontent.com/3902022/39715138-3b574420-51fa-11e8-83bd-6baa6787ecab.png">

**Comment by Tresky** [comment]
@alexandrudima Thank you for combining these together.

**Comment by RoyHP** [comment]
This bug is driving our team nuts. We have always expected the block within the brackets where the cursor is placed to be highlighted.

In our opinion, the highlight should be at the blue arrow. I personally don't know of any other editors that do it this way; it's very counterintuitive

**Comment by julesgilson** [comment]
doesn't seem logical to me either - maybe there are people who prefer it the way it is - if so then an additional setting to select parent or current would suit all. I would imagine they are the minority though..

**Comment by kryptxy** [comment]
Same issue. It's very confusing in HTML as well.
Screenshot
![vscode](https://user-images.githubusercontent.com/28785742/41642386-c4875822-7485-11e8-99e1-efadbc84c684.png)

I'd expect the `<div id="container">` indent line to be highlighted. Instead `<body>` tag's indent line is being highlighted.

**Comment by alexandrudima** [comment]
There are more good examples in https://github.com/Microsoft/vscode/issues/53737#issue-339017155

**Comment by hassanbinsafdar** [comment]
@alexandrudima you are saying everywhere this is duplicate i visited all references and in all refrences you are saying the same thing but not telling the solution 👎

**Comment by ericdrobinson** [comment]
@hassanbinsafdar With respect to your comments:
>you are saying everywhere this is duplicate

He said that for the actual duplicates, including [the one that I opened](https://github.com/Microsoft/vscode/issues/53737). **_This_** issue (#49342), you will note, is left in the **Open** state to track the problem.

>in all refrences you are saying the same thing but not telling the solution 

Currently, **there _is_ no solution**. It's an **Open** issue and is currently [in the Backlog](https://github.com/Microsoft/vscode/milestone/8), effectively marking it for "future fix" consideration.

The only "solution" available today is to disable the feature in your Settings:

```js
// Controls whether the editor should highlight the active indent guide
"editor.highlightActiveIndentGuide": false,
```

_[See [here](https://code.visualstudio.com/docs/getstarted/settings) for more details.]_

You could then use a third-party extension like [Bracket Pair Colorizer](https://marketplace.visualstudio.com/items?itemName=CoenraadS.bracket-pair-colorizer) to get the feature.

**Comment by quicksnap** [comment]
The current behavior is not intuitive to me as well. 

Bracket Pair Colorizer extension does work as I would expect, and has some great features (such as underlining the statement which defines the scope). Unfortunately it causes severe input lag when moving the cursor vertically very quickly.

**Comment by mrjones2014** [comment]
@CoenraadS I've actually installed your bracket pair colorizer 2 and it does, in fact, solve the problem well enough that I'm satisfied, for most editors. however, it doesn't solve the issue, for example, in HTML markup (since there's not really any "brackets" for it to highlight). would it be possible to use your extension to enable the indent guides for HTML/Vue single file component templates?

**Comment by quicksnap** [comment]
The UX provided by @CoenraadS's extension, IMO, is something that should be in core VSCode. It's _really_ awesome.

**Comment by mrjones2014** [comment]
@quicksnap  I agree, the way the built-in indent guides work has been my biggest gripe/annoyance since day 1. @CoenraadS's extension has stopped my from pulling my hair out on many an occasion. The built-in indent guides are _really_ not how's I'd expect or want them to work.

**Comment by shimondoodkin** [comment]
I have a similar issue with javascript, on recent update today

**Comment by julesgilson** [comment]
7 months later... yawn. Even if someone deems this to be the "correct" functionality then it seems plenty of others do not - just make it a setting.

**Comment by rigwild** [comment]
The `Bracket Pair Colorizer` does not support Vue single file components. Always got confused by this tiny indent guide not being on the correct indentation (I mean, as i expect it to be).
When you're scrolling through some lines expecting to visually match the ending tag it gets confusing.

**Comment by julesgilson** [comment]
> The `Bracket Pair Colorizer` extension does not support Vue single file components.

Bracket pair colorizer works OK for me in Vue components. See attached

![clipboard01](https://user-images.githubusercontent.com/10134933/52701687-ae62d280-2f72-11e9-8b35-ccfae3bbc59c.jpg)

**Comment by lllopo** [comment]
This is issue is driving me nuts too. Guys, this is simply so badly not intuitive. The idea is to highlight the current scope after all and currently it highlights ... what ?! All of the examples above are really good and valid, including the HTML one. If there are some (crazy) people that like the current behavior outhere, just add a setting, so everyone can set it they want and let's get it done.

**Comment by hassanbinsafdar** [comment]
@lllopo microsoft not want to leave his legacy They start anything with very good but with passage of time they start to annoy users I was quiet happy but now i switch to webstorm cause of these types of problems

**Comment by mrjones2014** [comment]
@lllopo yeah its getting pretty ridiculous that this isn't fixed..... I'm holding out hope, but I may just have to switch if this isn't fixed soon.

**Comment by quicksnap** [comment]
Everyone, please keep the comments constructive. VSCode team does a great job with the product. Nothing is stopping any of you from creating a PR. In the mean time, the aforementioned extension is available: marketplace.visualstudio.com/items?itemName=CoenraadS.bracket-pair-colorizer-2

Being negative toward the VSCode team doesn't encourage them to improve this feature.

**Comment by lllopo** [comment]
@quicksnap  I don't really understand how I have been 'negative' towards the team, as you state. I fully appreciate the great work the VS Code team does overall. It is a great product and actually the best of class free product IMO (maybe only SublimeText is coming close, but it's not really free). And just because of that,  having such an essential functionality working in such a weird way (actually not working) for such a long time is totally strange, having that I've seen many other, peripheral issues to get much more love from the team than this one. The simple truth is : the bigger the group of people expressing their dissatisfaction with the current way of working - the bigger the chance of something actually get moving (just like you did a few comments above, btw ;-) ). Simple as that and that's what I did.
As of bracket pair colorizer 2 - it is a nice one, but it's a no go, of course. I have a few small issues with it, annoying, but doesn't worth mentioning and there is the big one - it doesn't support HTML. Full stop.

**Comment by santaclaus21** [comment]
I'm considering going back to the previous editor because of this

**Comment by rikusen0335** [comment]
Nothing updates on this? I'm thinking vscode is awesome, but this issue is VERY IMPORTANT to work with html. Unfortunately, I'm considering going back to another editor too...

**Comment by kanlukasz** [comment]
I miss so much with this function. This is one good functions that Notepad++ have. It REALLY improve coding workflow. Please make it,i beg 

![obraz](https://user-images.githubusercontent.com/30685349/57297274-809fac80-70cf-11e9-87f4-b00f9c66d6b5.png)
![obraz](https://user-images.githubusercontent.com/30685349/57297279-84cbca00-70cf-11e9-9cad-ae065b154601.png)

**Comment by malonnnn** [comment]
Just leaving my 2 cents that this is still present in mid july and also driving me insane. going to try CoenraadS pair colorizer 2

**Comment by LadyCailin** [comment]
I was going to perhaps take a stab at this, but is this an acceptable change? If so, can this be marked with help-wanted or bug?

**Comment by kanlukasz** [comment]
>
> I'm having trouble getting your plugin to work.. does it only work with certain langauges?
>

Please, don't mix cases. This is VSCode repo, not Bracket Pair.
If you have a problem with CoenraadS plugin, open ticket in his repository
Keeping tickets clean is good for all of us  ❤️ 

---

> 
> I was going to perhaps take a stab at this, but is this an acceptable change? If so, can this be marked with help-wanted or bug?
>

I hope it is possible/acceptable and VS Team will implement that.


---

The Bracket Pair plugin is not a perfect solution because:
- it does not recognize semantics in tags (html div, etc.)
- it has a lot of problems with `Extension causes high cpu load`

Another screen from N++
![obraz](https://user-images.githubusercontent.com/30685349/61290477-3f610280-a7cc-11e9-99c7-21e845043d18.png)

**Comment by jabacchetta** [comment]
This, in combination with #58713 provides for some unintuitive code navigation.

**Comment by MatteCrystal** [comment]
I also think the current functionally in vs code is unacceptable especially since there seems to be no fix for html ATM which is what I need it for. Please add this feature, this is a deal breaker for many people.

**Comment by wmertens** [comment]
There's over 5000 open issues in this repo. It's not helpful to keep saying the feature is crucial for your particular workflow. Just add your 👍 on the parent comment, that is something the team can [search on too](https://github.com/microsoft/vscode/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).

So if you came here to say "me too", please refrain from doing so, it is counterproductive.

**Comment by LadyCailin** [comment]
I completed a pull request 2 months ago that mostly fixed this. Can I get some feedback on that?

**Comment by reneolivo** [comment]
I really hate this! I hope there was some way to configure it. Leaving a comment in hopes that this is fixed.

**Comment by kanlukasz** [comment]
Wow. Did I miss something, or the VS Team did something with this?
At the moment I tested it and it looks as it working as we expect (more or less)

Example 1:
![11](https://user-images.githubusercontent.com/30685349/76783673-9cc1af00-67b2-11ea-8fcd-b0a3f7b7e7f3.gif)

Example 2:
![22](https://user-images.githubusercontent.com/30685349/76783684-a21ef980-67b2-11ea-84e4-395e7060b636.gif)

**Comment by kanlukasz** [comment]
Wow. Did I miss something, or the VS Team did something with this?
At the moment I tested it and it looks as it working as we expect (more or less)

![indent](https://user-images.githubusercontent.com/30685349/76783905-03df6380-67b3-11ea-827e-8db4d3fcd012.gif)

**Comment by quicksnap** [comment]
Woah! What?! You're right @kanlukasz it's fixed now!

**Comment by LadyCailin** [comment]
Yep! This is 90% fixed now with this PR: https://github.com/microsoft/vscode/pull/77762 This bug still isn't closed, because the PR doesn't actually do semantic detection (it's just based on the indentation of the above and below lines) so there will still be edge cases where it doesn't work, but for the vast majority of cases and codebases, it will work just fine. See the PR for details about what cases it won't work in.

**Comment by mrjones2014** [comment]
@LadyCailin I've been watching this issue for almost 2 years. you are my hero!
